libbeat/common/mapstr.go

```diff
@@ -315,10 +315,45 @@ func tryToMapStr(v interface{}) (MapStr, bool) {
 	}
 }
 
-// walkMap walks the data MapStr to arrive at the value specified by the key.
+// walkMapRecursive walks the data MapStr to arrive at the value specified by the key.
 // The key is expressed in dot-notation (eg. x.y.z). When the key is found then
 // the given mapStrOperation is invoked.
-func walkMap(key string, data MapStr, op mapStrOperation) (interface{}, error) {
+func walkMapRecursive(key string, data MapStr, op mapStrOperation) (interface{}, error) {
+
+	// Splits up the key in two parts: full key and first part before the dot
+	keyParts := strings.SplitN(key, ".", 2)
+	_, exists := data[key]
+
+	// If leave node or key exists directly
+	if len(keyParts) == 1 || exists {
```

Comment:
This if could be split up in 2 parts so in case the `key` matches directly, no SplitN is needed.

Reply:
I was thinking this too. (But I don’t have any idea how common the case is.)

Reply:
Probably depends on us how we document things :-) I did split it up now as it brings some improvements in performance.

Reply:
Unfortunately we can't compare this part to the old behaviour as it does not exist there.